### PR TITLE
Extracting shared components

### PR DESCRIPTION
**Pull request by shakyShane**
Extracting some shared components for re-use in other pages.

The extracted things here are all used in the next PR, this was separated to enable easier review in isolation. (ie: it can be merged without affecting production code)

Main changes:
- EnvironmentProvider.js + environment.js
  - these can be used to create a consistent environment lookup
- ErrorBoundary.js
  -  just a default error boundary
- TranslationsProvider.js
  - a minimal translation implementation

https://app.asana.com/0/1201141132935289/1207666610467025/f

### CODE REVIEW

**Comment by shakyShane** [comment]
* **#979** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/979?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#983** <a href="https://app.graphite.dev/github/pr/duckduckgo/content-scope-scripts/983?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @shakyShane and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/special-pages/shared/components/EnvironmentProvider.js [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import { createContext, h } from 'preact'
+import { useContext, useEffect, useState } from 'preact/hooks'
+
+const EnvironmentContext = createContext({
+    isReducedMotion: /** @type {boolean} */(false),
+    isDarkMode: /** @type {boolean} */(false),
+    debugState: /** @type {boolean} */(false),
+    platform: /** @type {import('../environment').Environment['platform']} */('windows'),
+    willThrow: /** @type {boolean} */(false)
+})
```
Comment (shakyShane) [inline packages/special-pages/shared/components/EnvironmentProvider.js:10]: this represents the values that the application will care about.

File: packages/special-pages/shared/components/EnvironmentProvider.js [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+        const mediaQueryList = window.matchMedia(REDUCED_MOTION_QUERY)
+        const listener = (e) => setReducedMotion(e.matches)
+        mediaQueryList.addEventListener('change', listener)
+
+        // toggle events on window
+        window.addEventListener('toggle-reduced-motion', () => {
+            setReducedMotion(true)
+            document.documentElement.dataset.reducedMotion = String(true)
+        })
+        return () => mediaQueryList.removeEventListener('change', listener)
```
Comment (shakyShane) [inline packages/special-pages/shared/components/EnvironmentProvider.js:46]: This is used in platform UI tests, to speed up test execution

File: packages/special-pages/shared/components/TranslationsProvider.js [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+                })
+            })
+        })
+        return () => {
+            cleanupsCurr.forEach((fn) => fn())
+        }
+    }, [values, str])
+
+    return <span ref={ref} dangerouslySetInnerHTML={{ __html: str }} />
+}
```
Comment (shakyShane) [inline packages/special-pages/shared/components/TranslationsProvider.js:79]: This exists for the follow use-case: A translation might look like this:

```
"stringName": "Click here to go to <a>Settings</a>"
```

and then in a Preact context you can provide attributes or handlers as needed, such as:

```jsx
<Trans
    str={'Click here to go to <a>Settings</a>'}
    values={{
        a: {
            href: 'about:preferences',
            click: onSettingsHandler
        }
    }}
/>
```

This means we don't need to `innerHTML` any attributes at all

**Review by mgurgel — APPROVED** [review]
👍 Approved

**Pull request merged**